src/main.c

```diff
@@ -431,12 +431,15 @@ int main(int argc, char* argv[]) {
             fprintf(stderr, "%s: --indent takes one parameter\n", progname);
             die();
           }
-          dumpopts &= ~(JV_PRINT_TAB | JV_PRINT_INDENT_FLAGS(7));
-          int indent = atoi(argv[i+1]);
-          if (indent < -1 || indent > 7) {
+          char* end = NULL;
+          errno = 0;
+          long indent = strtol(argv[i+1], &end, 10);
+          if (errno || indent < -1 || indent > 7 ||
+              isspace(*argv[i+1]) || end == NULL || *end) {
```

Comment:
I have noticed two problems here:
1) `end == NULL` doesn't make sense, it seems entirely unnecessary
2) I think there should also be `end == argv[i+1]` as error condition otherwise `--indent ''` is accepted and treated as equivalent of `--indent 0`. Did you do this intentionally?